From 1994b018d6cc3c93aa678072eb4f901a84945ddc Mon Sep 17 00:00:00 2001
From: Johannes Thoma <johannes@johannesthoma.com>
Date: Fri, 8 Feb 2019 22:01:56 +0100
Subject: [PATCH 5/5] tcp_transport: Always return empty buffer on
 MSG_DONTWAIT.

Currently don't know how to implement this in Windows. It does
not hurt, since DRBD retries with the MSG_DONTWAIT flag unset.

We might move this patch into the internal wsk wrapper functions.
---
 drbd/drbd_transport_tcp.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drbd/drbd_transport_tcp.c b/drbd/drbd_transport_tcp.c
index 25be149..7eca7cf 100644
--- a/drbd/drbd_transport_tcp.c
+++ b/drbd/drbd_transport_tcp.c
@@ -322,6 +322,14 @@ static int dtt_recv(struct drbd_transport *transport, enum drbd_stream stream, v
 	if (!socket)
 		return -ENOTCONN;
 
+		/* If this should be non-blocking, pretend that we
+		 * haven't received anything. DRBD then will redo
+		 * the call without this flag set.
+		 */
+
+	if (flags & MSG_DONTWAIT)
+		return 0;
+
 	if (flags & CALLER_BUFFER) {
 		buffer = *buf;
 		rv = dtt_recv_short(socket, buffer, size, flags & ~CALLER_BUFFER);
-- 
2.17.0

